src/frontend/src/flows/manage/index.ts

```diff
@@ -271,6 +282,20 @@ export const renderManage = async ({
 
 export const displayManagePage = renderPage(displayManageTemplate);
 
+function isPinAuthenticated(
+  devices_: DeviceData[],
+  connection: AuthenticatedConnection
+): boolean {
+  const principal = connection.identity.getPrincipal();
+  const currentDevice = devices_.find((device) => {
+    const principal1 = Principal.selfAuthenticating(
+      new Uint8Array(device.pubkey)
+    );
+    return principal1.compareTo(principal) === "eq";
```

Comment:
is that more effective than comparing their textual representation?

`principal1.toText() === ....`

Reply:
I saw the compare method when I tried using `===` on the principals directly. But I agree, `toText` is clearer / nicer to read.